### PR TITLE
50 update rest adapter

### LINKED ISSUE #50: Update RESTApiDatabaseAdapter
Aaron has updated his model and started a hosting service. Our plugin needs to be connected so we can start publishing our data to the actual Database

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #50 

I have updated the `RESTApiDatabaseAdapter` to use the new endpoints on the API of the Octopeer server. If you want to manually test these changes, you will have to update your local Octopeer server as well.
Main changes:
- The new endpoints of the API are used, using names to query instead of IDs.
  - This caused the session identifier field and its belongings to be removed
- Use the new `URLHandler` to process a PR URL into repo owner/name/pr, instead of having them hard-coded in the constructor
  - Still missing: fetching the username from the context. In some way @aaronang still thinks about connecting to the GitHub API, forcing the user to log in authorize their GitHub in order to be able to post to the Database. In that case, we can easily fetch the username from this context.
- Added a script that can be executed (unfortunately only in the console of the database itself, but it's something :D) in order to fill the database with valid EventIDs and ElementIDs
- `started_at` is now a Unix Timestamp instead of a Date. In `IEventObject` it's in milliseconds (because that's the format of `new Date().time()`, and it's converted to seconds before posting it to the database.
- Updated tests (yay coverage! :D)

If there is anything you'd still like to have when posting a semantic-event to the database, let me know! In the mean time I'll continue working on #64 :) Have fun reviewing!

### CODE REVIEW

**Comment by MathiasMeuleman** [comment]
This looks nice! Tnx for the PR, woud be nice to test everything with a real database :)

Ready for merge

File: src/octopeer-github/main/DatabaseAdaptable/ConstantDatabasePoster.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Created by Maarten on 25-05-2016.
+ *
+ * Performs a series of POST requests in order to fill the constant part of the Database.
+ */
+
+// tslint:disable-next-line:no-unused-variable
+function constantDatabasePoster() {
```
Comment (mdingjan) [inline src/octopeer-github/main/DatabaseAdaptable/ConstantDatabasePoster.ts]: Is there a way of covering this function with a test case?

**Comment by mdingjan** [comment]
I've reviewed your changes; they are nice. :)
I've left one comment related to testing. Besides, from #74 there is still one more task related to testing. 
As soon as this is resolved, and dev is merged into this branch, this PR is ready for merging. :)

File: src/octopeer-github/main/DatabaseAdaptable/ConstantDatabasePoster.ts (hunk 1 above)
Comment (mpsijm) [inline src/octopeer-github/main/DatabaseAdaptable/ConstantDatabasePoster.ts]: No, there is not, because this relies on posting to the real database.

**Comment by mdingjan** [comment]
Thanks for resolving my feedback. :)
You've nicely tested the equals method for comparing arrays in ts. :)
Ready for merging.

**Comment by mdingjan** [comment]
Will merge!

**Pull request merged**